subsys/net/lib/prometheus/pm_formatter.c

```diff
@@ -0,0 +1,218 @@
+/*
+ * Copyright (c) 2024 Mustafa Abdullah Kus, Sparse Technology
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/net/prometheus/pm_formatter.h>
+#include <zephyr/net/prometheus/pm_collector.h>
+#include <zephyr/net/prometheus/pm_metric.h>
+#include <zephyr/net/prometheus/pm_histogram.h>
+#include <zephyr/net/prometheus/pm_summary.h>
+#include <zephyr/net/prometheus/pm_gauge.h>
+#include <zephyr/net/prometheus/pm_counter.h>
+
+#include <stdarg.h>
+#include <stdio.h>
+#include <string.h>
+
+#include <zephyr/kernel.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(pm_formatter, CONFIG_PROMETHEUS_LOG_LEVEL);
+
+static int write_to_buffer(char *buffer, size_t buffer_size, const char *str)
+{
+	/* helper function to write formatted string to buffer, handling buffer size */
+	int len = snprintf(buffer + strlen(buffer), buffer_size - strlen(buffer), "%s", str);
+
+	if (len >= 0 && len < buffer_size - strlen(buffer)) {
+		return 1;
+	}
+
+	/* error (insufficient buffer size) */
+	return 0;
+}
+
+static int write_metric_to_buffer(char *buffer, size_t buffer_size, const char *format, ...)
+{
+	/* helper function to write formatted metric to buffer */
+	char temp_buffer[256];
```

Comment:
This is quite large and will use quite large amount of stack, do we need this long tmp buffer?